test/e2e/instascale_test.go

```diff
@@ -0,0 +1,248 @@
+package e2e
+
+import (
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	. "github.com/project-codeflare/codeflare-operator/test/support"
+	mcadv1beta1 "github.com/project-codeflare/multi-cluster-app-dispatcher/pkg/apis/controller/v1beta1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestInstascale(t *testing.T) {
+
+	test := With(t)
+	test.T().Parallel()
+
+	namespace := test.NewTestNamespace()
+
+	// Test configuration
+	config := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: corev1.SchemeGroupVersion.String(),
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mnist-mcad",
+			Namespace: namespace.Name,
+		},
+		BinaryData: map[string][]byte{
+			// pip requirements
+			"requirements.txt": ReadFile(test, "mnist_pip_requirements.txt"),
+			// MNIST training script
+			"mnist.py": ReadFile(test, "mnist.py"),
+		},
+		Immutable: Ptr(true),
+	}
+	config, err := test.Client().Core().CoreV1().ConfigMaps(namespace.Name).Create(test.Ctx(), config, metav1.CreateOptions{})
+	test.Expect(err).NotTo(HaveOccurred())
+	test.T().Logf("Created ConfigMap %s/%s successfully", config.Namespace, config.Name)
+
+	// create OCM connection
+	instascaleOCMSecret, err := test.Client().Core().CoreV1().Secrets("default").Get(test.Ctx(), "instascale-ocm-secret", metav1.GetOptions{})
+	if err != nil {
+		test.T().Errorf("unable to retrieve instascale-ocm-secret - Error : %v", err)
+	}
```

Comment:
The test is expecting to be executed in OSD environment, right?

Reply:
Yes, I'm testing on OSD, ROSA and OCP which all use this secret.

Reply:
I believe OCP doesn't need the secret so will have to refactor to take that into account.